consensus/polybft/signer/public.go

```diff
@@ -6,36 +6,46 @@ import (
 	"math/big"
 
 	"github.com/0xPolygon/polygon-edge/helper/common"
-	bn256 "github.com/umbracle/go-eth-bn256"
+	"github.com/kilic/bn254"
 )
 
 // PublicKey represents bls public key
 type PublicKey struct {
-	p *bn256.G2
+	p *bn254.PointG2
 }
 
 // aggregate adds the given public keys
-func (p *PublicKey) aggregate(onemore *PublicKey) *PublicKey {
-	var g2 *bn256.G2
-	if p.p == nil {
-		g2 = new(bn256.G2).Set(&zeroG2)
-	} else {
-		g2 = new(bn256.G2).Set(p.p)
+func (p *PublicKey) aggregate(next *PublicKey) *PublicKey {
```

Comment:
As a separate task could we remove this code duplication(publicKey, signature)?

Reply:
At first glance it seemed like it can be unified into a single implementation, but eventually it cannot. The reason is because signature aggregation works with G1 point, whereas public key aggregates G2 points.